custom/icds_reports/reports/prevalence_of_undernutrition.py

```diff
@@ -208,10 +208,15 @@ def get_prevalence_of_undernutrition_data_chart(domain, config, loc_level, show_
         data['red'][date_in_miliseconds]['unweighed'] += (total - weighed)
         data['red'][date_in_miliseconds]['weighed'] += weighed
 
-    top_locations = sorted(
-        [dict(loc_name=key, percent=value) for key, value in six.iteritems(best_worst)],
-        key=lambda x: x['percent']
-    )
+    all_locations = [
+        {
+            'loc_name': key,
+            'percent': value
+        }
+        for key, value in six.iteritems(best_worst)
+    ]
+    all_locations_sorted_by_name = sorted(all_locations, key=lambda x: x['loc_name'])
+    all_locations_sorted_by_percent_and_name = sorted(all_locations_sorted_by_name, key=lambda x: x['percent'])
```

Comment:
is there anyway to do this without sorting twice? it doesnt seem like the first value is used at all.

Reply:
The goal is to specify a deterministic sort order (by name) when the percents are equal.  This is required for tests python 2/3 changes the default dictionary iteration order.

```all_locations_sorted_by_name``` is used as an input to the second sort that creates ```all_locations_sorted_by_percent_and_name```.

re: double sort - I'm going to do a second pass on all these changes (and those in https://github.com/dimagi/commcare-hq/pull/24116/) to implement this recommendation where possible: https://github.com/dimagi/commcare-hq/pull/24116/files/f7741d55e7ad87de96d093f6db2cc9de8a415cef#diff-7231be0c95659b29a34b182f3aa8e8b5

Reply:
:+1: